targets/target-mbed/source/jerry_port.c

```diff
@@ -0,0 +1,141 @@
+/* Copyright 2014-2015 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#define _BSD_SOURCE
+#include <stdarg.h>
+#include <stdlib.h>
+#include <sys/time.h>
+
+#include "jerry-core/jerry-port.h"
+
+#include "mbed-hal/us_ticker_api.h"
+
+static bool abort_on_fail = false;
+
+/**
+ * Provide log message to filestream implementation for the engine.
+ */
+int jerry_port_logmsg (FILE *stream, /**< stream pointer */
+                       const char *format, /**< format string */
+                       ...) /**< parameters */
+{
+  va_list args;
+  int count;
+  va_start (args, format);
+  count = vfprintf (stream, format, args);
+  va_end (args);
+  return count;
+} /* jerry_port_logmsg */
+
+/**
+ * Provide error message to console implementation for the engine.
+ */
+int jerry_port_errormsg (const char *format, /**< format string */
+                         ...) /**< parameters */
+{
+  va_list args;
+  int count;
+  va_start (args, format);
+  count = vfprintf (stderr, format, args);
+  va_end (args);
+  return count;
+} /* jerry_port_errormsg */
+
+/**
+ * Provide output character to console implementation for the engine.
+ */
+int jerry_port_putchar (int c) /**< character to put */
+{
+  return putchar ((unsigned char) c);
+} /* jerry_port_putchar */
+
+/**
+ * Sets whether 'abort' should be called instead of 'exit' upon exiting with
+ * non-zero exit code in the default implementation of jerry_port_fatal.
+ */
+void jerry_port_default_set_abort_on_fail (bool flag) /**< new value of 'abort on fail' flag */
```

Comment:
Copying over the `*_abort_on_fail` functions from the default port implementation seems to be a misunderstanding. From main.cpp and jerry_run.cpp, it does not seem as if you needed an "do an exit or an abort on fail?" decision. The default port extends the main jerry API with these functions and logic so that the jerry command line interpreter on Linux or OSX can control fatal exit behaviour via command line options. On the boards, however, you can predetermine that. So, I'd suggest drop these two functions and implement `jerry_port_fatal` in a way that simply either calls `abort` or `exit`.

(BTW, the `jerry_port_default_` prefix of these functions hints that they are specific to the default port.)